packages/samples/headless-react/src/pages/DependentFacet.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  buildFacet,
+  buildSearchEngine,
+  Facet as HeadlessFacet,
+  getSampleSearchEngineConfiguration,
+  loadFacetSetActions,
+  SearchEngine,
+} from '@coveo/headless';
+import {Component} from 'react';
+import {Facet} from '../components/facet/facet.fn';
+import {AppContext} from '../context/engine';
+
+export class DependentFacet extends Component<
+  {},
+  {visibleDependantFacet: boolean}
+> {
+  private engine: SearchEngine;
+  private facets: HeadlessFacet[];
+  constructor(props: {}) {
+    super(props);
+    this.engine = buildSearchEngine({
+      configuration: getSampleSearchEngineConfiguration(),
+    });
+    const fields = ['objecttype', 'author', 'source', 'filetype'];
+    this.facets = fields.map((field) =>
+      buildFacet(this.engine, {options: {field}})
+    );
```

Comment:
Could the sample be clearer (for education purposes) if we created two properties, `parentFacet` and `dependentFacet`, vs. looping through an array?

This could remove the need for the getters on lines 79-85 too,